avocado/utils/configure_network.py

```diff
@@ -35,111 +35,176 @@ class NWException(Exception):
     """
 
 
-def set_ip(ipaddr, netmask, interface, interface_type=None):
+class Network_interface:
     """
-    Gets interface name, IP, subnet mask and creates interface
-    file based on distro.
+    class For network interface
     """
-    if distro.detect().name == 'rhel':
-        conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s" % interface
-        if os.path.exists(conf_file):
-            shutil.move(conf_file, conf_file+".backup")
-        else:
-            raise NWException("%s interface not available" % interface)
-        with open(conf_file, "w") as network_conf:
-            if interface_type is None:
-                interface_type = 'Ethernet'
-            network_conf.write("TYPE=%s \n" % interface_type)
-            network_conf.write("BOOTPROTO=none \n")
-            network_conf.write("NAME=%s \n" % interface)
-            network_conf.write("DEVICE=%s \n" % interface)
-            network_conf.write("ONBOOT=yes \n")
-            network_conf.write("IPADDR=%s \n" % ipaddr)
-            network_conf.write("NETMASK=%s \n" % netmask)
-            network_conf.write("IPV6INIT=yes \n")
-            network_conf.write("IPV6_AUTOCONF=yes \n")
-            network_conf.write("IPV6_DEFROUTE=yes")
-
-        cmd = "ifup %s" % interface
+
+    def __init__(self, interface):
+        self._interface = interface
+
+    def set_ip(self, ipaddr, netmask, interface_type=None):
```

Comment:
Multiline docstrings should follow PEP257:

https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

We should keep the params but we need a "summary" line and a "good long description".